.gitignore

```diff
@@ -1,23 +1,10 @@
-/artifacts
 /foundry_artifacts
 cache
-coverage
-report
-lcov.info
-build
-deployments/localhost
-deployments/*/solcInputs/*.json
-dist
 downloads
-export
-node_modules
-res
-src/types
-target
 test/uarch-bin
 test/uarch-log
-.node*
 .vscode
-*.abi
 *.log
-/out
+test/UArchReplay_*.t.sol
+src/UArchConstants.sol
+src/UArchStep.sol
```

Comment:
Should we also ignore the whole `ready_src/` dir?

Reply:
One thing nice to have the `ready_src` committed is that when someone wants to use this project as a submodule, it' ready right away. If we ignore the directory, he'll have to generate it manually. What do you think?

Reply:
Good point! On the one hand, it's generated stuff (just run `make ...`). It feels like good practice not adding it to git. On the other hand, it can be quite practical. I don't know haha, I'll let you decide :)

Reply:
I think I'll keep `ready_src/` checked-in in github. So when people are trying to using this library, they simply add it as a submodule, and don't have to worry about dependencies like `emulator, Make, Docker`...
Also, if the person accidentally modified the library source code, (either manually or with script), it may be hard to debug without git telling that the source tree has been changed.